### DISCUSSION TITLE
Note記法内のコードブロックに `:::` があるとNoteが終了してしまう

### DISCUSSION THREAD

**Discussion by yarakos95** [increments/qiita-discussions#306] [要望・アイデア・不具合]
Note 記法内にある Indented code block や Fenced code block に `:::` が含まれていると、Note の終了とみなされてしまいます。

Fenced code block では、Note の外を出てコードブロックになってしまいます。

## Indented code block のサンプル

![image](https://user-images.githubusercontent.com/75878492/191663737-ea7b2f0c-e16c-409c-90ab-be431ce9b241.png)

<details><summary>サンプルコード</summary>

```markdown
:::note info
Indented code block に `:::` だけが含まれている。

    :::

ここは Note の中です。
:::
```

</details>

-----

![image](https://user-images.githubusercontent.com/75878492/191663819-bcd8e28e-2da1-40d4-b5c5-c00e0c250188.png)

<details><summary>サンプルコード</summary>

```markdown
:::note info
Indented code block に `:::` が含まれている。

    print("Hello World")
    :::

ここは Note の中です。
:::
```

</details>

-----

![image](https://user-images.githubusercontent.com/75878492/191663959-d0f552c7-e698-4a76-a6b6-793234f30fbd.png)

<details><summary>サンプルコード</summary>

```markdown
:::note info
Indented code block に `:::` が含まれている。

    :::
    print("Hello World")

これは Note の中です。
:::
```

</details>

## Fenced code block のサンプル

![image](https://user-images.githubusercontent.com/75878492/191660910-a30576e3-12eb-4f75-9b20-e5d56821ef35.png)

<details><summary>サンプルコード</summary>

````markdown
:::note info
Fenced code block に `:::` だけが含まれる。

```
:::
```

:::
````

</details>

-----

![image](https://user-images.githubusercontent.com/75878492/191661534-6ebef2d7-d02f-4fb6-a293-534d53060cc3.png)

<details><summary>サンプルコード</summary>

````markdown
:::note info
Fenced code block に `:::` が含まれる。

```python
print("Hello World")
:::
```

:::

ここまでコードブロックになります。
どこまでのコードブロックになります。
````

</details>

-----

![image](https://user-images.githubusercontent.com/75878492/191661273-8fa03318-c63e-490b-af12-f253091c5b2a.png)

<details><summary>サンプルコード</summary>

````markdown
:::note info
Fenced code block に `:::` が含まれる。

```python
:::
print("Hello World")
```

:::

ここまでコードブロックになります。
どこまでのコードブロックになります。
````

</details>

## 追記

どうやら、コードブロックに限らず、行頭に `:::` が来る場合におかしくなってしまうようです。

引用符では、2 行目以降に `>` を省略することが出来ますが、ここに `:::` があると同様の問題を生じます。

![image](https://user-images.githubusercontent.com/75878492/192144044-cf73a880-69a0-4a7f-a114-256536d044e5.png)

<details><summary>サンプルコード</summary>

```markdown
:::note info

> 引用
:::
ここも引用のはず。

ここは Note の中のはず。
:::
```

</details>

-----

複数行に渡って行内コードを書く場合でも、やはり同様の結果を得ます。

![image](https://user-images.githubusercontent.com/75878492/192144519-85024417-88a4-4158-b0e2-624247a2a597.png)

<details><summary>サンプルコード</summary>

```markdown
:::note info
複数行に分けて `code` を書きます。
``
code
:::
code
``

ここは Note 内です。
:::
```

</details>

-----

数式内でも同じ結果を得ます。

![image](https://user-images.githubusercontent.com/75878492/192145230-9e472296-de8a-46ec-9bbe-e4741bd1fb64.png)

<details><summary>サンプルコード</summary>

```markdown
:::note info

$$
f(x) = ax^2 + bx + c
:::
$$

ここは Note 内です。
:::
```

</details>

また、表の 1 列目左の `|` を省略して `:::` を書いても同じことが起こります。

**Comment by tanaka-asami415** [comment]
@Yarakashi-Kikohshi 
詳細に検証結果をありがとうございます！🙏
前回ご報告いただいた内容で issue https://github.com/increments/qiita-discussions/issues/304 を作成していますので、こちらと合わせて調査いたします🙇
